app/src/routes/transfer-new/+page.svelte

```diff
@@ -0,0 +1,306 @@
+import { userAddrCosmos } from "$lib/wallet/cosmos"
+import Chevron from "./(components)/chevron.svelte"
+import { userBalancesQuery } from "$lib/queries/balance"
+import * as Card from "$lib/components/ui/card/index.ts"
+import type { Chain, UserAddresses } from "$lib/types.ts"
+import { Input } from "$lib/components/ui/input/index.js"
+import { userAddrOnChain } from "$lib/utilities/address.ts"
+import { Button } from "$lib/components/ui/button/index.ts"
+import ChainDialog from "./(components)/chain-dialog.svelte"
+import ChainButton from "./(components)/chain-button.svelte"
+import AssetsDialog from "./(components)/assets-dialog.svelte"
+import { getSupportedAsset } from "$lib/utilities/helpers.ts"
+import { debounce, raise, sleep } from "$lib/utilities/index.ts"
+import ArrowLeftRight from "virtual:icons/lucide/arrow-left-right"
+import { transferSchema, type TransferSchema } from "./validation.ts"
+import { getCosmosChainInfo } from "$lib/wallet/cosmos/chain-info.ts"
+import { submittedTransfers } from "$lib/stores/submitted-transfers.ts"
+import { parseUnits, formatUnits, type HttpTransport, getAddress } from "viem"
+import { aptosStore, userAddressAptos, getAptosWallet } from "$lib/wallet/aptos"
+import { cosmosStore, getCosmosOfflineSigner } from "$/lib/wallet/cosmos/config.ts"
+import { type Writable, writable, derived, get, type Readable } from "svelte/store"
+import { custom, switchChain, getConnectorClient, waitForTransactionReceipt } from "@wagmi/core"
+
+type SearchParams = { [key: string]: string }
+
+const querClient = useQueryClient()
+
+function queryData<T extends Array<unknown>>(
+  key: Array<string>,
+  filter?: (value: T[number]) => boolean
+): T {
+  const data = querClient.getQueryData<T>(key) ?? []
+  return (filter ? data.filter(filter) : data) as T
+}
+
+let chains = queryData<Array<Chain>>(["chains"], chain => chain.enabled_staging)
+
+let transferQueryOptions = queryOptions<SearchParams>({
+  enabled: query => false,
+  placeholderData: keepPreviousData,
+  queryKey: ["transfer-state", $page.url.toString()],
+  initialData: Object.fromEntries($page.url.searchParams),
+  queryFn: ({ queryKey, signal, meta }) => Object.fromEntries($page.url.searchParams)
+})
+
+let state = createQuery(transferQueryOptions)
+
+const observer = new QueryObserver<SearchParams>(querClient, {
+  enabled: query => false,
+  queryKey: ["transfer-state"]
+})
+
+let userAddress = derived(
+  [userAddrCosmos, userAddrEvm, userAddressAptos],
+  ([$cosmos, $evm, $aptos]) => ({ evm: $evm, aptos: $aptos, cosmos: $cosmos })
+)
+
+$: asset = $page.url.searchParams.get("asset") || $state.data?.asset
+$: amount = $page.url.searchParams.get("amount") || $state.data?.amount
+$: receiver = $page.url.searchParams.get("receiver") || $state.data?.receiver
+$: source = $page.url.searchParams.get("source") || $state.data?.source
+$: destination = $page.url.searchParams.get("destination") || $state.data?.destination
```

Comment:
What do tehse lines do? it seems that currently `asset` is defined in three places:

1. `searchParams.get("asset")`
2. `searchParams.data?.asset`
3. `$: asset = ...`

Let's make sure we have only 1, strongly-typed, place where de fine the user input that is to be validated.

If we need to initially set this manually based on some initial data from the URL, I suggest handling this in an onMount call rather than introducing reactive state